bentoml/service/artifacts/__init__.py

```diff
@@ -105,6 +118,14 @@ def wrapped_pack(*args, **kwargs):
                         "`pack` an artifact multiple times may lead to unexpected "
                         "behaviors"
                     )
+                if 'metadata' in kwargs:
+                    if isinstance(kwargs['metadata'], dict):
+                        self._metadata = kwargs['metadata']
+                    else:
+                        logger.warning(
```

Comment:
What's the reason for doing a warning here instead of raising a TypeError here?  It seems to me it's better to raise an error explicitly, and help the user to discover this mistake early